src-django/api/serializer.py

```
@@ -48,6 +49,13 @@ def to_representation(self, instance):
 
         return ret
 
+    def validate(self, data):
+        if 'element_type' in data and data['element_type'] in models.Element.CHOICE_TYPES:
```

also if it's not in `CHOICE_TYPES`, it should also throw an error